catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1467,7 +1467,7 @@ public QueryResponse query(QueryRequest queryRequest, FederationStrategy strateg
         } catch (RuntimeException re) {
             LOGGER.warn("Exception during runtime while performing query", re);
             throw new UnsupportedQueryException("Exception during runtime while performing query");
-
+            //TODO -- Why transform all runtime exceptions to unsupported query exceptions?
```

Comment:
@stustison I agree; however, there was an interesting case where this was getting thrown due to an issue with the executor. It was misleading because the query was supported just fine. Is there anything we can do at this juncture?